packages/auth-remix/src/server.ts

```diff
@@ -0,0 +1,795 @@
+      | ((
+          params: ParamsOrError<{ tokenData: TokenData }>
+        ) => Response | Promise<Response>),
+    cb?: (
+      params: ParamsOrError<{ tokenData: TokenData }>
+    ) => Response | Promise<Response>
+  ): Promise<
+    | Response
+    | {
+        tokenData: TokenData;
+        headers: Headers;
+      }
+  > {
+    return handleAction(
+      async (data, headers, req) => {
+        const verifier = parseCookies(req)[this.options.pkceVerifierCookieName];
+
+        if (!verifier) {
+          throw new Error("no pkce verifier cookie found");
+        }
+
+        let resetToken = new URL(req.url).searchParams.get("reset_token");
+
+        if (Array.isArray(resetToken)) {
+          resetToken = resetToken[0];
+        }
+
+        if (!resetToken) {
+          throw new Error("reset token not found");
+        }
+
+        const [password] = _extractParams(
+          data,
+          ["password"],
+          "password missing"
+        );
+        const tokenData = await (
+          await this.core
+        ).resetPasswordWithResetToken(resetToken, verifier, password);
+
+        headers.append(
+          "Set-Cookie",
+          `${this.options.authCookieName}=${tokenData.auth_token}; HttpOnly; SameSite=Lax`
+        );
+
+        headers.append("Set-Cookie", `${this.options.pkceVerifierCookieName}=`);
+
+        return { tokenData };
+      },
+      req,
+      dataOrCb,
+      cb
+    );
+  }
+
+  async signout(
+    cb?: () => Response | Promise<Response>
+  ): Promise<{ headers: Headers } | Response> {
+    const headers = new Headers({
+      "Set-Cookie": `${
+        this.options.authCookieName
+      }=; HttpOnly; SameSite=strict; expires=${Date.now()}`,
+    });
+
+    if (cb) return callbackCall(cb, headers);
+
+    return { headers };
+  }
+
+  async createUser(tokenData: TokenData, provider?: BuiltinProviderNames) {
+    let username: string | null = null;
+    if (tokenData.provider_token && provider === "builtin::oauth_github") {
+      const { data } = await new Octokit({
+        auth: tokenData.provider_token,
+      }).request("get /user");
+
+      username = data.login;
+    }
+    await this.client.query(
+      `
+    with identity := (select ext::auth::Identity filter .id = <uuid>$identity_id),
+    insert User {
+      identity := identity,
+      name := <optional str>$username ?? (
+        select ext::auth::EmailFactor filter .identity = identity
+      ).email
+    } unless conflict on .identity`,
+      {
+        identity_id: tokenData.identity_id,
+        username: username,
+      }
+    );
+  }
+}
+
+function parseCookies(request: Request) {
+  const cookies = request.headers.get("Cookie");
+
+  return cookies
+    ? cookies.split(";").reduce((cookies, cookie) => {
+        const [name, val] = cookie.split("=");
+        cookies[name.trim()] = val.trim();
+        return cookies;
+      }, {} as { [key: string]: string | undefined })
+    : {};
+}
```

Comment:
As above, we should probably replace this with a proper cookie parsing library like:  https://www.npmjs.com/package/cookie